### DISCUSSION TITLE
Support Plugin Capability - iPXE

### DISCUSSION THREAD

**Discussion by xunholy** [siderolabs/sidero#773] [Ideas]
@frezbo and myself have been delving deep into the weeds to discover why RockPis and potentially other SBCs won't work with iPXE.

We found that there are some bugs and limitations with their firmware and how u-boot with iPXE handles the CPU sleep cycle.

More information about that in this thread https://forum.radxa.com/t/rock-pi-4c-ipxe/8554/10

To fix this we will need to patch sidero however this is an anti-pattern as ideally sidero core should be untouched with a specific use case or code for a particular machine. Rather it would make sense to allow for sidero to be extensible so that certain functionality could be bolted on without also having to maintain a fork of sidero.

I could see this working potentially somewhat like Talos Linux extentions but again I'm open to other solutions so that we can easily support an array of different hardware/firmware combinations with minimal upkeep.

**Comment by smira** [comment]
yep, having some way to extend Sidero would be really nice. It can be not only different iPXE binaries, but also different Sidero agent environemnts, etc.
